CRM/Core/ManagedEntities.php

```diff
@@ -77,6 +77,16 @@ public function reconcile($modules = NULL) {
     $modules = $modules ? (array) $modules : NULL;
     $declarations = $this->getDeclarations($modules);
     $plan = $this->createPlan($declarations, $modules);
+    if (!CRM_Core_Config::isUpgradeMode()) {
+      $plan = $this->optimizePlan($plan);
```

Comment:
Note to self: The optimization here is related to the new test-failures on `cv`'s `ExtensionLifecycleTest`. It took a while to track/confirm. To reproduce, I have to run two-part test on `cv`, eg

```bash
./scripts/run-tests.sh --debug tests/Command/ExtensionLifecycleTest.php --filter '(testLifecycleWithShortNames|testDownloadWithExplicitUrl)'
```

(*I'm a bit tired from hunting it down -- but that gives something better to focus on...*)